src/storage/log/aof/log.rs

```diff
@@ -146,11 +146,10 @@ impl Aol {
 
         // Check if the record is larger than the maximum file size
         if rec.len() > self.opts.max_file_size as usize {
-            eprintln!("rec len {} {}", rec.len(), self.opts.max_file_size);
             return Err(Error::RecordTooLarge);
         }
 
-        let _lock = self.mutex.write();
+        let _lock = self.mutex.lock();
```

Comment:
I would suggest to rename _lock, lock and use drop later when you want to be sure the lock id dropped.
Otherwise I suspect that a compiler optimisation may release the lock early.

Reply:
I wasn't sure myself at first, but after some research, I realized this is a valid usage!
A variable definition automatically adds a drop at the scope end regardless of whether it's prefixed with _ or not.
The compiler won't remove it as an optimization because, if it did, it would change the program's observable behavior (lock taken/released).

Reply:
Regardless, I second the idea of using e.g. RwLock as Emmanuel has suggested, because it makes us not to accidentally miss locking when mutating active_segment.

Reply:
Thanks @emmanuel-keller and @mumoshu for the details! I will tackle this in a separate PR to make lock exclusive for the active segment